### PR TITLE
Add rake task for load users and balances

### PR DESCRIPTION
**Pull request by mnaichuk**

### CODE REVIEW

File: lib/tasks/import.rake [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  # - email
+  #
+  # Make sure that you create required currency
+  # Usage:
+  # For import users: -> bundle exec rake import:users['file_name.csv']
+
+  desc 'Load members from csv file.'
+  task :users, [:config_load_path] => [:environment] do |_, args|
+    csv_table = File.read(Rails.root.join(args[:config_load_path]))
+    errors_users_file = File.open("errors_users_file.txt", "w")
```
Comment (ysv) [inline lib/tasks/import.rake]: write to /log/import_something.log

File: lib/tasks/import.rake [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    count = 0
+    CSV.parse(csv_table, headers: true).map do |row|
```
Comment (ysv) [inline lib/tasks/import.rake]: use each

File: lib/tasks/import.rake [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      email = row['email']
```
Comment (ysv) [inline lib/tasks/import.rake]:
```ruby
:role=>:member, :state=>:active} 
2.6.3 :002 > row = {email: 'kek@lol.com', uid: '1234', level: 1}
 => {:email=>"kek@lol.com", :uid=>"1234", :level=>1} 
2.6.3 :003 > row.reverse_merge(defaults)
 => {:level=>1, :role=>:member, :state=>:active, :email=>"kek@lol.com", :uid=>"1234"} 
```

File: lib/tasks/import.rake [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      uid = row['uid']
+      level = row.fetch('level', 0)
+      role = row.fetch('role', 'member')
+      state = row.fetch('state', 'active')
+      Member.create!(uid: composed_uid, email: email, level: level, role: role, state: state)
+      count += 1
+    rescue => e
```
Comment (ysv) [inline lib/tasks/import.rake]: use StandardError

File: lib/tasks/import.rake [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    csv_table = File.read(Rails.root.join(args[:config_load_path]))
+    errors_users_file = File.open("errors_users_file.txt", "w")
+    count = 0
+    CSV.parse(csv_table, headers: true).map do |row|
+      email = row['email']
+      uid = row['uid']
+      level = row.fetch('level', 0)
+      role = row.fetch('role', 'member')
+      state = row.fetch('state', 'active')
+      Member.create!(uid: composed_uid, email: email, level: level, role: role, state: state)
```
Comment (ysv) [inline lib/tasks/import.rake]: don't forget about composed_uid

File: lib/tasks/import.rake [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+      count += 1
+    rescue => e
+      message = { error: e.message, email: row['email'], uid: row['uid'] }
+      errors_users_file.write(message.to_yaml + "\n")
+    end
+    errors_users_file.close
+    Kernel.puts "Created #{count} members"
```
Comment (ysv) [inline lib/tasks/import.rake:30]: Optional: Add also number of errored members

File: lib/tasks/import.rake [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+  desc 'Load accounts balances from csv file.'
+  task :accounts, [:config_load_path] => [:environment] do |_, args|
+    csv_table = File.read(Rails.root.join(args[:config_load_path]))
+    errors_accounts_file = File.open("errors_accounts_file.txt", "w")
+    count = 0
+    CSV.parse(csv_table, headers: true).map do |row|
+      uid = row['uid'] 
+      member = Member.find_by_uid!(uid)
+      currency = Currency.find(row['currency_id'])
+      account = Account.find_or_create_by!(member: member, currency_id: currency)
```
Comment (ysv) [inline lib/tasks/import.rake]: use currency: currency

File: lib/tasks/import.rake [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+      locked_balance = row['locked_balance'].to_d
+      next if main_balance <= 0 && locked_balance <= 0
+
+      asset_code = currency.coin? ? 102 : 101
+      main_liability_code, locked_liability_code = currency.coin? ? [202, 212] : [201, 211]
+      ActiveRecord::Base.transaction do
+        Operations::Asset.create!(code: asset_code, currency_id: currency.id, credit: main_balance + locked_balance)
+        Operations::Liability.create!(code: main_liability_code, currency_id: currency.id, member_id: member.id, credit: main_balance) if main_balance > 0
+        Operations::Liability.create!(code: locked_liability_code, currency_id: currency.id, member_id: member.id, credit: locked_balance) if locked_balance > 0
+        account&.update!(balance: main_balance, locked: locked_balance)
```
Comment (ysv) [inline lib/tasks/import.rake]: remove &

File: lib/tasks/import.rake [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+        count += 1
+      end
+    rescue => e
+      message = { error: e.message, accoutn_id: row['AccountId'], composed_uid: composed_uid }
```
Comment (ysv) [inline lib/tasks/import.rake]: uid

File: lib/tasks/import.rake [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+      currency = Currency.find(row['currency_id'])
+      account = Account.find_or_create_by!(member: member, currency_id: currency)
+      main_balance = row['main_balance'].to_d
+      locked_balance = row['locked_balance'].to_d
+      next if main_balance <= 0 && locked_balance <= 0
+
+      asset_code = currency.coin? ? 102 : 101
+      main_liability_code, locked_liability_code = currency.coin? ? [202, 212] : [201, 211]
+      ActiveRecord::Base.transaction do
+        Operations::Asset.create!(code: asset_code, currency_id: currency.id, credit: main_balance + locked_balance)
```
Comment (ysv) [inline lib/tasks/import.rake]: use `credit!` instead of create!

**Review by ysv — CHANGES_REQUESTED** [review]

File: docs/databases/import.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
+## How to import users and account balances database Peatio
```
Comment (ysv) [inline docs/databases/import.md]: May be we move to doc/tasks

File: docs/databases/import.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+
+1. Create `csv` files for users and accounts with those templates.
+
+### Users table
+
+  |      uid      | email          | level |    role      |  state  | referral_id  |
```
Comment (ysv) [inline docs/databases/import.md]: Do we actually need referral_id ?
If yes probably need to be renamed to referral_uid

File: lib/tasks/import.rake [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+  # Usage:
+  # For import users: -> bundle exec rake import:users['file_name.csv']
+
+  desc 'Load members from csv file.'
+  task :users, [:config_load_path] => [:environment] do |_, args|
+    csv_table = File.read(Rails.root.join(args[:config_load_path]))
+    import_user_log = File.open("./log/import_users.log", "w")
+    count = 0
+    CSV.parse(csv_table, headers: true).each do |row|
+      row = row.to_hash.symbolize_keys!.delete_if { |_k, v| v.nil? }
```
Comment (ysv) [inline lib/tasks/import.rake]: `row.to_h.symbolize_keys.compact` - looks better for me

may be even move it to map:
```ruby
CSV.parse(csv_table, headers: true).map(&:to_h).map(&:symbolize_keys) ...
```

Also check maybe it's possible to symbolize or parse step I think there should be some option
may be even compact option

**Review by ysv — CHANGES_REQUESTED** [review]

File: docs/databases/import.md (hunk 12 above)
Comment (mnaichuk) [reply docs/databases/import.md]: Yes, I want create template for peatio and barong. Agreed, will change to referral_uid

**Review by ysv — APPROVED** [review]

**Pull request merged**